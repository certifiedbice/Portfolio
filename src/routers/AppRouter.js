import React,{Component} from 'react';
import {Switch,Route} from 'react-router-dom';
import Landing from '../pages/landing/Landing';
import Home from '../pages/home/Home';
import About from '../pages/about/About';
import Contact from '../pages/contact/Contact';
import Portfolio from '../pages/portfolio/Portfolio';
import Resume from '../pages/resume/Resume';
import Services from '../pages/services/Services';
import NotFound from '../pages/notfound/NotFound';
export default class AppRouter extends Component{
	render(){
		return(
			<Switch>
				<Route exact path='/' component={Landing}/>
				<Route exact path='/home' component={Home}/>
				<Route exact path='/about' component={About}/>
				<Route exact path='/contact' component={Contact}/>
				<Route exact path='/portfolio' component={Portfolio}/>
				<Route exact path='/resume' component={Resume}/>
				<Route exact path='/services' component={Services}/>
				<Route component={NotFound}/>
			</Switch>
		);
	}
}