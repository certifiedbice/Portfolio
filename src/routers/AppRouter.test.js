import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import AppRouter from './AppRouter'

it('renders without crashing',()=>{
    const root=document.createElement('root');
    ReactDOM.render(<BrowserRouter><AppRouter/></BrowserRouter>,root);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<BrowserRouter><AppRouter/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});