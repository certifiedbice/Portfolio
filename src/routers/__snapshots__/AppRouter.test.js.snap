// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    id="background-image-container"
  >
    <div
      id="background-image"
    />
  </div>,
  <div
    id="landing-container"
  >
    <div
      id="portrait-overlay"
    />
    <div
      id="landing-message"
    >
      Uncompromisingly different
    </div>
    <div
      id="proceed-message"
    >
      Click or tap somewhere...
    </div>
  </div>,
  <div
    id="page-fade"
  />,
]
`;
